packages/ui/src/proposals/modals/AddNewProposal/components/SpecificParameters/UpdateWorkingGroupBudget.tsx

```diff
@@ -0,0 +1,99 @@
+import BN from 'bn.js'
+import React, { useEffect } from 'react'
+
+import { InputComponent, InputNumber } from '@/common/components/forms'
+import { Info } from '@/common/components/Info'
+import { Row } from '@/common/components/Modal'
+import { RowGapBlock } from '@/common/components/page/PageContent'
+import { TextInlineMedium, TextMedium } from '@/common/components/typography'
+import { BN_ZERO } from '@/common/constants'
+import { capitalizeFirstLetter } from '@/common/helpers'
+import { useNumberInput } from '@/common/hooks/useNumberInput'
+import { formatTokenValue } from '@/common/model/formatters'
+import { SelectWorkingGroup } from '@/working-groups/components/SelectWorkingGroup'
+import { useWorkingGroup } from '@/working-groups/hooks/useWorkingGroup'
+import { GroupIdName } from '@/working-groups/types'
+
+export interface UpdateWorkingGroupBudgetParameters {
+  budgetUpdate?: BN
+  groupId?: GroupIdName
+}
+
+interface UpdateWorkingGroupBudgetProps extends UpdateWorkingGroupBudgetParameters {
+  setBudgetUpdate: (amount: BN) => void
+
+  setGroupId(groupId: string): void
+}
+
+export const UpdateWorkingGroupBudget = ({
+  budgetUpdate,
+  setBudgetUpdate,
+  groupId,
+  setGroupId,
+}: UpdateWorkingGroupBudgetProps) => {
+  const [amount, setAmount] = useNumberInput(0, budgetUpdate)
+
+  const { group } = useWorkingGroup({ name: groupId })
+
+  const isDisabled = !group
+
+  useEffect(() => {
+    setBudgetUpdate(new BN(amount))
+  }, [amount])
+  useEffect(() => {
+    setBudgetUpdate(BN_ZERO)
+  }, [groupId])
```

Comment:
I think it is correct now because it's not group budget but the update only. So when another group is selected the update input value should change to zero and then the user can decide whether to increase or decrease the budget.
The current group budget is handled elsewhere.